plugins/optimization-detective/optimization.php

```diff
@@ -216,10 +206,23 @@ function od_optimize_template_output_buffer( string $buffer ): string {
 	 */
 	do_action( 'od_register_tag_visitors', $tag_visitor_registry );
 
+	$visitors     = iterator_to_array( $tag_visitor_registry );
+	$current_etag = implode( ',', array_keys( $visitors ) );
```

Comment:
An important point here: similar to the `slug`, i think the etag actually should be hashed:

```suggestion
	$current_etag = md5( implode( ',', array_keys( $visitors ) ) );
```

The key reason here is that when expanding the scope here to address the parent issue #1466, there will be much more data to include in the ETag. So I suggest that there be a new function created, like `od_compute_current_etag()` which takes `$tag_visitor_registry` as an argument. For now it could look like:

```php
function od_compute_current_etag( OD_Tag_Visitor_Registry $tag_visitor_registry ) {
    $data = array(
        'tag_visitors' => array_keys( iterator_to_array( $tag_visitor_registry ) ),
    );
    return md5( serialize( $data ) );
}
```

Then later when working on #1466 the `$data` can be expanded to include things like the IDs and `post_modified` times for the posts in the loop, what the current queried object is, and so on.

Reply:
Doen in 8fe38f7.